input-profile.ts

```diff
@@ -0,0 +1,369 @@
+     * @param {Object3D} obj - The object to retrieve components from.
+     * @return {Component[]} An array of components.
+     * @hidden
+     */
+    private _getComponents(obj: Object3D) {
+        if (obj == null) return;
+        const components: Component[] = [];
+        const stack = [obj];
+        while (stack.length > 0) {
+            const currentObj = stack.pop();
+            const comps = currentObj
+                .getComponents()
+                .filter((c: Component) => !this.toFilter.has(c.type));
+            components.push(...comps);
+            const children = currentObj.children || [];
+            // Push children onto the stack in reverse order to maintain the correct order
+            for (let i = children.length - 1; i >= 0; --i) {
+                stack.push(children[i]);
+            }
+        }
+        return components;
+    }
+
+    /**
+     * Activates or deactivates components based on the specified boolean value.
+     * @param {boolean} active - If true, components are set to active; otherwise, they are set to inactive.
+     * @hidden
+     */
+    private _setComponentsActive(active: boolean) {
+        const comps = this._defaultControllerComponents;
+        if (comps == undefined) return;
+        for (let i = 0; i < comps.length; ++i) {
+            comps[i].active = active;
+        }
+    }
+
+    /**
+     * Event handler triggered when XR input sources change.
+     * Detects new XR input sources and initiates the loading of input profiles.
+     * @param {XRInputSourceChangeEvent} event - The XR input source change event.
+     * @hidden
+     */
+    private _onInputSourcesChange(event: XRInputSourceChangeEvent) {
+        if (this._isModelLoaded() && !this.mapToDefaultController) {
+            this._setComponentsActive(false);
+        }
+
+        event.added.forEach((xrInputSource: XRInputSource) => {
+            if (xrInputSource.hand != null) return;
+            if (this._handedness != xrInputSource.handedness) return;
+
+            this.gamepad = xrInputSource.gamepad;
+            const profile =
+                this.customProfileFolder !== ''
+                    ? this.customProfileFolder
+                    : this.path + xrInputSource.profiles[0];
+            this.url = profile + '/profile.json';
+
+            this._profileJSON = InputProfile.Cache.has(this.url)
+                ? InputProfile.Cache.get(this.url)
+                : null;
+
+            if (this._profileJSON != null) return;
+            fetch(this.url)
+                .then((res) => res.json())
+                .then((out) => {
+                    this._profileJSON = out;
+                    InputProfile.Cache.set(this.url, this._profileJSON);
+
+                    if (!this._isModelLoaded()) this._loadAndMapGamepad(profile);
+                })
+                .catch((e) => {
+                    console.error('failed to load profile from' + this.url);
+                    console.error(e);
```

Comment:
```suggestion
                    console.error('Failed to load profile from', this.url, e);
```

Reply:
Probably need a word in between to make the transition with the original error tho, otherwise that looks like that:

<img width="363" alt="image" src="https://github.com/WonderlandEngine/components/assets/8783766/55adff29-1d47-4523-88b3-8816e07c4ed7">

Maybe:

```ts
console.error('Failed to load profile from', this.url, 'reason:', e);
```

Reply:
:thumbsup: